halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/kubernetes/KubernetesAccountValidator.java

```diff
@@ -0,0 +1,156 @@
+    } catch (IOException e) {
+      psBuilder.addProblem(ERROR, e.getMessage());
+      return;
+    }
+
+    if (context != null && !context.isEmpty()) {
+      Optional<NamedContext> namedContext = kubeconfig
+          .getContexts()
+          .stream()
+          .filter(c -> c.getName().equals(context))
+          .findFirst();
+
+      if (!namedContext.isPresent()) {
+        psBuilder.addProblem(ERROR, "Context \"" + context + "\" not found in kubeconfig \"" + kubeconfigFile + "\"")
+            .setRemediation("Either add this context to your kubeconfig, rely on the default context, or pick another kubeconfig file");
+        smoketest = false;
+      }
+    } else {
+      String currentContext = kubeconfig.getCurrentContext();
+      if (currentContext.isEmpty()) {
+        psBuilder.addProblem(ERROR, "You have not specified a Kubernetes context, and your kubeconfig \"" + kubeconfigFile + "\" has no current-context")
+            .setRemediation("Either specify a context in your halconfig, or set a current-context in your kubeconfig");
+        smoketest = false;
+      } else {
+        psBuilder.addProblem(WARNING, "You have not specified a Kubernetes context in your halconfig, Spinnaker will use \"" + kubeconfigFile + "\" instead")
+            .setRemediation("We recommend explicitly setting a context in your halconfig, to ensure changes to your kubeconfig won't break your deployment");
+      }
+    }
+
+    if (smoketest) {
+      Config config = KubernetesConfigParser.parse(kubeconfigFile, context, cluster, user, namespaces);
+      KubernetesClient client = new DefaultKubernetesClient(config);
+
+      try {
+        client.namespaces().list();
+      } catch (Exception e) {
+        psBuilder.addProblem(ERROR, "Unable to communicate with your Kubernetes cluster: " + e.getMessage())
+            .setRemediation("Verify that these credentials work manually using \"kubectl\"");
+      }
+    }
+  }
+
+  private void validateDockerRegistries(ProblemSetBuilder psBuilder, KubernetesAccount account, DeploymentConfiguration deployment) {
+    List<DockerRegistryReference> dockerRegistries = account.getDockerRegistries();
+    List<String> namespaces = account.getNamespaces();
+
+    if (dockerRegistries == null || dockerRegistries.isEmpty()) {
+      psBuilder.addProblem(ERROR, "You have not specified any docker registries to deploy to")
+          .setRemediation("Add a docker registry that can be found in this deployment's dockerRegistries provider");
+    }
+
+    DockerRegistryProvider dockerRegistryProvider = deployment.getProviders().getDockerRegistry();
+    if (dockerRegistryProvider == null || dockerRegistryProvider.getAccounts() == null || dockerRegistryProvider.getAccounts().isEmpty()) {
+      psBuilder.addProblem(ERROR, "The docker registry provider has not yet been configured for this deployment")
+          .setRemediation("Kubernetes needs a Docker Registry as an image source to run");
```

Comment:
Actually, I think it's best to hold off on that, since ideally the registry should be added with halyard, but there is no documentation for that. I'll add a todo.